stdlib/optparse.pyi

```diff
@@ -99,25 +101,46 @@ class Option:
     ACTIONS: tuple[str, ...]
     ALWAYS_TYPED_ACTIONS: tuple[str, ...]
     ATTRS: list[str]
-    CHECK_METHODS: list[Callable[..., Incomplete]] | None
+    CHECK_METHODS: list[Callable[[Self], object]] | None
     CONST_ACTIONS: tuple[str, ...]
     STORE_ACTIONS: tuple[str, ...]
     TYPED_ACTIONS: tuple[str, ...]
     TYPES: tuple[str, ...]
-    TYPE_CHECKER: dict[str, Callable[[Option, str, Incomplete], Any]]
+    TYPE_CHECKER: dict[str, Callable[[Option, str, str], Any]]
     _long_opts: list[str]
     _short_opts: list[str]
     action: str
+    type: str | None
     dest: str | None
-    default: Incomplete
+    default: Any
     nargs: int
-    type: Incomplete
+    const: Any | None
+    choices: list[str] | tuple[str, ...] | None
     callback: Callable[..., Incomplete] | None
     callback_args: tuple[Incomplete, ...] | None
     callback_kwargs: dict[str, Incomplete] | None
     help: str | None
     metavar: str | None
-    def __init__(self, *opts: str | None, **attrs) -> None: ...
+    def __init__(
+        self,
+        *opts: str | None,
+        # The following keywords are handled by the _set_attrs method. All default to
+        # `None` except for `default`, which defaults to `NO_DEFAULT`.
+        action: str | None = None,
+        type: str | builtins.type | None = None,
+        dest: str | None = None,
+        default: Any = ...,  # = NO_DEFAULT
+        nargs: int | None = None,
+        const: Any | None = None,
+        choices: list[str] | tuple[str, ...] | None = None,
+        # TODO: callback, callback_args, callback_kwargs must be all supplied or all omitted. Add overloads.
+        # Revisit if ParamSpec is ever changed to support non-unpacked args and kwargs.
+        callback: Callable[..., Incomplete] | None = None,
```

Comment:
Should these Incompletes be Any as the correct type cannot currently be represented in the type system?

Reply:
Maybe. Let’s change it later, after I have done some testing. I’m not done with optparse yet.